plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -83,6 +83,13 @@ default Optional<String> getNamespaceSeparator()
 
     List<TableInfo> listTables(ConnectorSession session, Optional<String> namespace);
 
+    default List<SchemaTableName> listIcebergTables(ConnectorSession session, Optional<String> namespace)
+    {
+        return listTables(session, namespace).stream()
```

Comment:
Do we need the default implementation ? It doesn't look like it would return only iceberg tables

Reply:
I don't, I thought it is a good default but since most of the catalogs override it anyway it makes sense to implement it explicitly everywhere.